lib/zclient.c

```diff
@@ -2370,6 +2370,7 @@ bool zapi_srv6_sid_notify_decode(struct stream *s, struct srv6_sid_ctx *ctx,
 			*p_locator_name = NULL;
 		else {
 			STREAM_GET(locator_name, s, len);
+			locator_name[len] = '\0';
```

Comment:
The issue here is that `locator_name` is declared as `static`, which means its value is preserved across function calls. As a result, the buffer is not automatically cleared between calls.
 
Wouldn't it be more readable to make `locator_name` non-static and initialise it to zero, instead of explicitly adding the string terminator?
 
This is the approach used in other functions such as `zclient_vrf_add`, `zclient_interface_add`, and `zebra_interface_state_read`.

Reply:
Actually, the interfaces taken as example do not return a pointer to a buffer.
I have to let static , otherwise, the returned pointer will be a dangling pointer.

what I can do to clarify is separate the memset from the pointer initialisation.